src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.cpp

```diff
@@ -407,9 +545,14 @@ void MultiDiracDeterminant::resize(int nel)
   psiV.resize(NumOrbitals);
   dpsiV.resize(NumOrbitals);
   d2psiV.resize(NumOrbitals);
+  dspin_psiV.resize(NumOrbitals);
   psiM.resize(nel, NumOrbitals);
   dpsiM.resize(nel, NumOrbitals);
   d2psiM.resize(nel, NumOrbitals);
+  dspin_psiM.resize(nel, NumOrbitals);
+  //psiM_temp.resize(nel,morb);
```

Comment:
all the other checks I've used have relied on the ParticleSet.is_spinor_ flag to check. This function doesn't have access to a particleset, so another solution is needed in order to only resize for spinors. 

My current solution is to add an argument to the constructor that sets an internal spinor flag inside MultiDiracDeterminant (default as false). Only need to add something like
```
std::vector<std::unique_ptr<MultiDiracDeterminant>> dets;
bool spinor = targetPtcl.is_spinor_;
for (int grp = 0; grp < nGroups; grp++)
{
  app_log() << "      Creating base determinant (" << grp << ") for MSD expansion. \n";
  dets.emplace_back(std::make_unique<MultiDiracDeterminant>(std::move(spo_clones[grp]), grp, spinor));
  }
```
to the SlaterDetBuilder::buildComponent() since it has access to the particleset when the MultiDiracDets are built. 

Once that flag is set,  I can just check if the internal flag in MultiDiracDeterminant is a spinor, and appropriately resize the arrays. Also, some assertions could be added inside the other functions used in evaluations that check this flag to make sure the ParticleSet  spinor flag agrees with the internal MultiDiracDeterminant flag set by the constructor. That would make sure nothing gets mismatched.

If this sounds fine, I'll make the changes. Otherwise, suggestions are welcome

Reply:
fixed